gui/wxpython/lmgr/frame.py

```diff
@@ -1122,16 +1122,30 @@ def OnChangeMapset(self, event):
                                             None,
                                             mapset)
 
-    def OnMapsetChanged(self, dbase, location, mapset):
+    def OnMapsetChanged(self, dbase, location, mapset, fallback_session):
         """Current mapset changed.
         If location is None, mapset changed within location.
+        In addition, if temporary db is not current, it deletes temporary location.
         """
         if not location:
             self._setTitle()
         else:
             # close current workspace and create new one
             self.OnWorkspaceClose()
             self.OnWorkspaceNew()
+        if fallback_session:
+            gisenv = grass.gisenv()
+            grassdb = os.environ["TMPDIR"]
+            location = cfg.temporary_location
+            if not grassdb == gisenv["GISDBASE"]:
+                # Delete temporary location
+                delete_location(grassdb, location)
+                self._giface.grassdbChanged.emit(
+                    location=location, grassdb=grassdb, action="delete", element="grassdb"
+                )
```

Comment:
Could we handle this in switch_mapset_interactively directly and not introduce the fallback_session parameter everywhere? Would then be the only problem the dismissing of the infobar? The info bar could connect the mapset changed signal and dismiss itself, but maybe I am missing something (possibly some problem with first time infobars?)

Reply:
I think that it is okay to dismiss infobar in case of mapset changes. I did it as you suggested. Just one thing - the second infobar for data import is not this case of mapset changed so now it is not being dismissed... This issue could be solved by third ShowImportSuccessfulInfo message in https://github.com/OSGeo/grass/pull/1205.